source/opt/instruction.cpp

```diff
@@ -470,5 +470,36 @@ bool Instruction::IsOpaqueType() const {
 
 bool Instruction::IsFoldable() const { return opt::IsFoldableOpcode(opcode()); }
 
+std::ostream& operator<<(std::ostream& str, const ir::Instruction& inst) {
+  // Convert the module to binary.
+  std::vector<uint32_t> module_binary;
+  inst.context()->module()->ToBinary(&module_binary, /* skip_nop = */ false);
+
+  // Convert the instruction to binary. This is used to identify the correct
+  // stream of words to output from the module.
+  std::vector<uint32_t> inst_binary;
+  inst.ToBinaryWithoutAttachedDebugInsts(&inst_binary);
+
+  // Do not generate a header.
+  uint32_t options = SPV_BINARY_TO_TEXT_OPTION_NO_HEADER;
```

Comment:
Interesting. From a debugging perspective I feel the raw ids are more valuable. How about a refactor as follows?

// In Instruction. Always or's in the NO_HEADER option.
std::string Instruction::PrettyPrint(uint32_t options) const;

Then change to the ostream operator to use pretty print as I've set it up (raw ids). This would give people option to change other options.